Contribution.md

```diff
@@ -0,0 +1,153 @@
+# Contribution guidelines
+The guidelines for contribution to babitz backend project are listed below. Please go through it before any contribution.
+
+## 1. General guidelines
+- The general guidelines for writing code. We will be using a subset of [typescript contribution guidelines](https://github.com/microsoft/TypeScript/wiki/Coding-guidelines). In case any of these conflict with loopback4 guidelines, loopback4's guideline is taken as priority. 
+
+  ### Names
+    1. Use PascalCase for type names.
+    2. Use PascalCase for enum values.
+    3. Use camelCase for function names.
+    4. Use camelCase for property names and local variables.
+    5. Do not use `_` as a prefix for private properties.
+    6. Use whole words in names when possible.
+
+  ### Types
+    1. Do not export types/functions unless you need to share it across multiple components.
+    2. Do not introduce new types/values to the global namespace.
+    3. Shared types should be defined in `types.ts`.
+    4. Within a file, type definitions should come first.
+
+  ### `null` and `undefined`
+    1. Use `undefined`. Do not use null.
+
+  ### General Assumptions
+    1. Consider objects like Nodes, Symbols, etc. as immutable outside the component that created them. Do not change them.
+    2. Consider arrays as immutable by default after creation.
+
+  ### Flags
+    1. More than 2 related Boolean properties on a type should be turned into a flag.
+
+  ### Comments
+    1. Use JSDoc style comments for functions, interfaces, enums, and classes.
+
+  ### General Constructs
+    1. Do not use `for..in` statements; instead, use `ts.forEach`, `ts.forEachKey` and `ts.forEachValue`. Be aware of their slightly different semantics.
+    2. Try to use `ts.forEach`, `ts.map`, and `ts.filter` instead of loops when it is not strongly inconvenient.
+
+  ### Style
+    1. Use arrow functions over anonymous function expressions.
+    2. Only surround arrow function parameters when necessary. <br />For example, `(x) => x + x` is wrong but the following are correct:
+        - `x => x + x`
+        - `(x,y) => x + y`
+        - `<T>(x: T, y: T) => x === y`
+    3. Always surround loop and conditional bodies with curly braces. Statements on the same line are allowed to omit braces.
+    4. Open curly braces always go on the same line as whatever necessitates them.
+    5. Parenthesized constructs should have no surrounding whitespace. <br />A single space follows commas, colons, and semicolons in those constructs. For example:
+        - `for (var i = 0, n = str.length; i < 10; i++) { }`
+        - `if (x < 10) { }`
+        - `function f(x: number, y: string): void { }`
+    6. Use a single declaration per variable statement <br />(i.e. use `var x = 1; var y = 2;` over `var x = 1, y = 2;`).
+    7. `else` goes on a separate line from the closing curly brace.
+    8. Use 1 tab per indentation.
```

Comment:
We are using `prettier` to take enforce style. Let's link the instruction of running the same here.

Reply:
done.